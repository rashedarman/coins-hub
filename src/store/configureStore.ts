import { configureStore } from '@reduxjs/toolkit';
import cryptoSlice from './crypto/cryptoSlice';
import newsSlice from './news/newsSlice';

const rootReducer = {
  crypto: cryptoSlice,
  news: newsSlice,
};

const store = configureStore({
  reducer: rootReducer,
});

export default store;
