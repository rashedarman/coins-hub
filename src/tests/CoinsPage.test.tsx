import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CoinsPage from '../pages/Coins';
import store from '../store/configureStore';

describe('Testing CoinsListTable component', () => {
  it('Renders CoinsListTable component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CoinsPage />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
