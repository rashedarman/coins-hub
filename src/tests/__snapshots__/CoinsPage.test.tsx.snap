// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoinsListTable component Renders CoinsListTable component correctly 1`] = `
<div>
  <div
    class="mantine-Container-root mantine-sz575t"
  >
    <div
      class="mantine-11ol2j9"
    >
      <h1
        class="mantine-Text-root mantine-Title-root mantine-10wkz9q"
      >
        Best Coin Price Tracker in the Market
      </h1>
      <div
        class="mantine-Text-root mantine-1pbxw0k"
      >
        With CoinStats, you can manage all your crypto assets from one interface.
      </div>
    </div>
    <div
      class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <div
        class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid="false"
          class="mantine-Input-input mantine-TextInput-input mantine-ywb7zt"
          id="mantine-r0"
          placeholder="Search crypto"
          type="text"
        />
        <div
          class="mantine-1n9cu5e mantine-Input-rightSection mantine-TextInput-rightSection"
        >
          <div />
        </div>
      </div>
    </div>
    <div
      class="mantine-6hoaow"
    >
      <svg
        class="mantine-1avyp1d"
        height="36px"
        role="presentation"
        stroke="#228be6"
        viewBox="0 0 38 38"
        width="36px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(2.5 2.5)"
          >
            <circle
              cx="16"
              cy="16"
              r="16"
              stroke-opacity=".5"
            />
            <path
              d="M32 16c0-9.94-8.06-16-16-16"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 16 16"
                repeatCount="indefinite"
                to="360 16 16"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
